### PR TITLE
Heartbeat scheduler

### PR DESCRIPTION
**Pull request by didovesei**

### CODE REVIEW

File: forch/heart_beat_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""Schedule heart beat with functions that have to be called in order"""
+import logging
+import threading
+
+LOGGER = logging.getLogger('HeartBeat')
```
Comment (grafnu) [inline forch/heart_beat_scheduler.py]: heartbeat  (all logging sections are lowercase by convention)

File: forch/heart_beat_scheduler.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class HeartBeatScheduler:
```
Comment (grafnu) [inline forch/heart_beat_scheduler.py]: HeartbeatScheduler -- heartbeat is one word (same with filename -- heartbeat_scheduler.py)

File: forch/heart_beat_scheduler.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    def _periodic_task(self):
+        for callback in self._callbacks:
+            try:
+                callback()
+            except Exception as error:
+                LOGGER.error("Running %s: %s", callback, error)
+
+        threading.Timer(self._interval, self._periodic_task).start()
+
+    def start(self):
```
Comment (grafnu) [inline forch/heart_beat_scheduler.py]: will need a stop method

File: forch/heart_beat_scheduler.py (hunk 1 above)
Comment (didovesei) [reply forch/heart_beat_scheduler.py]: Done

File: forch/heart_beat_scheduler.py (hunk 2 above)
Comment (didovesei) [reply forch/heart_beat_scheduler.py]: Done

File: forch/heart_beat_scheduler.py (hunk 3 above)
Comment (didovesei) [reply forch/heart_beat_scheduler.py]: Done

**Pull request merged**